lib/spack/spack/util/web.py

```diff
@@ -738,7 +739,8 @@ def find_versions_of_archive(
 
         # We'll be a bit more liberal and just look for the archive
         # part, not the full path.
-        url_regex = os.path.basename(url_regex)
+        # this is a URL so it is a posixpath even on Windows
+        url_regex = posixpath.basename(url_regex)
```

Comment:
We only support Python 3.7+ now, so lets use `Pathlib` over the `os.path` module. 
```suggestion
        url_regex = pathlib.PurePath(url_regex).name
```

Reply:
Done. I use PurePosixPath though - the same idea as using posixpath instead of path.